### DISCUSSION TITLE
Alerts use case: showing status messages that disappear when status changes

### DISCUSSION THREAD

**Discussion by jenniferarnesen** [dhis2/notes#252] [Q&A] [answered]
Dashboard currently uses alerts to notify user when a dashboard is loading and is taking a longer than usual. It sets up its own AlertStack component instead of using the app-platform's built in alert service.

The dashboard solution uses a timeout to delay the alert from showing for 500ms, and when loading is complete, the alert is removed from the DOM (the whole AlertStack).

I'm wondering if there is a way to achieve this behavior with the alert service?

**Comment by varl** [comment]
Hm, seems to me that we could expose `remove` together with the `show` when using the [`useAlert`](https://runtime.dhis2.nu/#/hooks/useAlert) hook, that would simplify this a bit.

Then we could delay the `show`, and set it as `permanent` so it doesn't disappear) and call `remove` when loading is complete.

**Reply by varl** [reply]
The only way to work around this today would be to use the [`useAlerts`](https://runtime.dhis2.nu/#/hooks/useAlerts) hook instead of `useAlert`.

**Reply by amcgee** [reply]
A `hide()` function for each alert was in the initial proposed implementation for `useAlert`, it was removed during iteration (@HendrikThePendric knows more) I believe because there wasn't a clear use-case for it.  I liked the symmetry of having it in the original implementation sketch I did, would be in favor of adding it back in

**Reply by amcgee** [reply]
There's an existing issue to track this - https://jira.dhis2.org/browse/LIBS-114

**Comment by varl** [comment, marked as answer]
https://jira.dhis2.org/browse/LIBS-114
